openslides_backend/action/mixins/delegation_based_restriction_mixin.py

```diff
@@ -3,10 +3,10 @@
 from openslides_backend.action.action import Action
 from openslides_backend.shared.filters import And, FilterOperator, Or
 
-from ....permissions.base_classes import Permission
-from ....permissions.permission_helper import has_perm
-from ....services.datastore.commands import GetManyRequest
-from ....shared.exceptions import MissingPermission
+from ...permissions.base_classes import Permission
+from ...permissions.permission_helper import has_perm
+from ...services.datastore.commands import GetManyRequest
+from ...shared.exceptions import MissingPermission
 
 DelegationBasedRestriction = Literal[
```

Comment:
1. IMO this are still user permissions and would be better to let them at their place in action/user than moving to a crowded place.
2. You should think about the check_delegator_restriction method performance. 
 - All your database calls are with locked fields. A permission check is made in one moment, next moment this could be changed. 
 - What do you think of a first db-read for meeting_user-collection with filter(OR(all meeting ids) AND user_id=given user_id. In the result you'll find all relevant meeting_ids and could check for the condition in meeting_user. The remaining meetings could be read for their restriction fields.

Reply:
Regarding point 1: I do believe I moved this because where it was it caused circular DI errors, so I don't think moving it back is a good idea. If you have an alternative place where you think this may fit better, feel free to tell me.

Regarding point 2: I don't understand what you mean, let's talk about it separately.

Reply:
Created new issue #2508